sound/soc/sof/intel/hda.c

```diff
@@ -53,67 +53,65 @@ static int sdw_clock_stop_quirks = SDW_INTEL_CLK_STOP_BUS_RESET;
 module_param(sdw_clock_stop_quirks, int, 0444);
 MODULE_PARM_DESC(sdw_clock_stop_quirks, "SOF SoundWire clock stop quirks");
 
+/* update the DAI config with the ALH stream_id. The IPC will be sent during FE hw_params */
```

Comment:
then why do you call out the FE hw_params in the comments?

Reply:
because Im removing the IPC from BE hw_params but the IPC will still need to be send and it will be sent during FE hw_params.

Reply:
what if you have two FEs mixed into a single BE? When would you send the IPC then?

In general anything that ties FEs and BEs is doomed to fail sooner or later.

Reply:
Yes, I would send the IPC if two FEs are mixed into a single BE and this should be OK because the FW should do nothing when the second IPC is sent with the widget being already active.

Reply:
We probably need to brainstorm a bit more. If there is a mixer, we need to have a fixed configuration for the mixer that does not depend on upstream components and hw_params. In addition, if there is an SRC behind the mixer, e.g. to downsample to a lower rate, the DAI configuration isn't necessarily the same as the mixer.

Reply:
@plbossart Im probably not seeing what you are in terms of potential problems with this change. But I want to reiterate that this PR does not change the existing functionality. It just moves the DAI_CONFIG IPC to a later point in time. With the BE hw_params being invoked before FE hw_params for us, I just have to make sure that the IPC for the BE DAI widgets is sent only after the widget is set up (which will be in FE hw_params). In terms of the mixer/src or any other widget config itself, this patch wont change anything.

Reply:
I will need a 1:1 to understand how the BE hw_params can be done before the FE hw_params, since the latter may use hw_params that affect the latter.

Also this DAI_CONFIG seems to be a problem across the board, it's handled differently from hw_params except when it's not.